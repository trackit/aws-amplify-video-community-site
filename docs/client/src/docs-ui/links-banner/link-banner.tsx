import {Component, h} from "@stencil/core";
import {
  containerInnerStyle,
  containerOuterStyle,
  logoStyle,
} from "./link-banner.style";
import * as links from "../../constants/links";
import * as img from "../../constants/img";

@Component({tag: "docs-link-banner", shadow: false})
export class DocsLinkBanner {
  render() {
    return (
      <docs-container
        class={containerOuterStyle}
        innerClass={containerInnerStyle}
      >
        <amplify-external-link href={links.GITHUB} graphic="black">
          <img src="/assets/github.svg" class={logoStyle} />
          Amplify GitHub
        </amplify-external-link>
        <amplify-external-link href={links.DISCORD} graphic="black">
          <img
            class={logoStyle}
            alt={img.DISCORD.alt}
            src={img.DISCORD.blueSrc}
          />
          Amplify on Discord
        </amplify-external-link>
        <amplify-external-link href={links.MARKETING} graphic="black">
          <img src="/assets/aws-dark.svg" class={logoStyle} />
          Amplify Resources
        </amplify-external-link>
        <amplify-external-link href={links.COMMUNITY} graphic="black">
          <img src="/assets/logo-dark.svg" class={logoStyle} />
          Amplify Community
        </amplify-external-link>
      </docs-container>
    );
  }
}
