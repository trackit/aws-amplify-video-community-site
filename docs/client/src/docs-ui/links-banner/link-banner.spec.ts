import {DocsLinkBanner} from "./link-banner";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-link-banner", () => {
  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsLinkBanner],
            html: `<docs-link-banner />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
