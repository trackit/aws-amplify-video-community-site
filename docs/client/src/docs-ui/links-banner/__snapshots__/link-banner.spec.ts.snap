// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-link-banner Render logic should render 1`] = `
<docs-link-banner>
  <docs-container class="css-1gfnuqu" innerclass="css-1ym5me9">
    <amplify-external-link graphic="black" href="https://github.com/aws-amplify">
      <img class="css-14cx9l" src="/assets/github.svg">
      Amplify GitHub
    </amplify-external-link>
    <amplify-external-link graphic="black" href="https://discord.gg/amplify">
      <img alt="Discord Logo" class="css-14cx9l" src="/assets/discord-blue.svg">
      Amplify on Discord
    </amplify-external-link>
    <amplify-external-link graphic="black" href="https://aws.amazon.com/amplify/framework/">
      <img class="css-14cx9l" src="/assets/aws-dark.svg">
      Amplify Resources
    </amplify-external-link>
    <amplify-external-link graphic="black" href="https://amplify.aws/community">
      <img class="css-14cx9l" src="/assets/logo-dark.svg">
      Amplify Community
    </amplify-external-link>
  </docs-container>
</docs-link-banner>
`;
